packages/wallets/src/features/cashier/modules/Transfer/hooks/useTransferMessages.ts

```diff
@@ -0,0 +1,292 @@
+        (exchangeRates?.rates[transferDirection === 'from' ? targetAccount.currency : sourceAccount.currency] ?? 1);
+    const availableSumConverted =
+        availableSumActiveWalletCurrency *
+        (exchangeRates?.rates[transferDirection === 'from' ? targetAccount.currency : sourceAccount.currency] ?? 1);
+
+    const sourceCurrencyLimit = transferDirection === 'from' ? allowedSumActiveWalletCurrency : allowedSumConverted;
+    const targetCurrencyLimit = transferDirection === 'from' ? allowedSumConverted : allowedSumActiveWalletCurrency;
+
+    const sourceCurrencyRemainder =
+        transferDirection === 'from' ? availableSumActiveWalletCurrency : availableSumConverted;
+    const targetCurrencyRemainder =
+        transferDirection === 'from' ? availableSumConverted : availableSumActiveWalletCurrency;
+
+    const formattedSourceCurrencyLimit = displayMoney?.(
+        sourceCurrencyLimit,
+        sourceAccount.currencyConfig.display_code,
+        sourceAccount.currencyConfig.fractional_digits
+    );
+    const formattedTargetCurrencyLimit = displayMoney?.(
+        targetCurrencyLimit,
+        targetAccount.currencyConfig.display_code,
+        targetAccount.currencyConfig?.fractional_digits
+    );
+
+    const formattedSourceCurrencyRemainder = displayMoney?.(
+        sourceCurrencyRemainder,
+        sourceAccount.currencyConfig.display_code,
+        sourceAccount.currencyConfig.fractional_digits
+    );
+    const formattedTargetCurrencyRemainder = displayMoney?.(
+        targetCurrencyRemainder,
+        targetAccount.currencyConfig?.display_code,
+        targetAccount.currencyConfig?.fractional_digits
+    );
+
+    if (availableSumActiveWalletCurrency === 0)
+        return {
+            text: `You've reached the lifetime transfer limit from your ${sourceAccount.accountName} to any ${targetWalletType} Wallet. Verify your account to upgrade the limit.`,
+            type: 'error' as const,
+        };
+
+    if (allowedSumActiveWalletCurrency === availableSumActiveWalletCurrency)
+        return {
+            text: `The lifetime transfer limit from ${sourceAccount.accountName} to any ${targetWalletType} Wallet is ${formattedSourceCurrencyLimit} (${formattedTargetCurrencyLimit}).`,
+            type: sourceAmount > sourceCurrencyRemainder ? ('error' as const) : ('success' as const),
+        };
+
+    return {
+        text: `Remaining lifetime transfer limit is ${formattedSourceCurrencyRemainder} (${formattedTargetCurrencyRemainder}). Verify your account to upgrade the limit.`,
+        type: sourceAmount > sourceCurrencyRemainder ? ('error' as const) : ('success' as const),
+    };
+};
+
+const cumulativeAccountLimitsMessageFn = ({
+    displayMoney,
+    exchangeRates,
+    limits,
+    sourceAccount,
+    sourceAmount,
+    targetAccount,
+}: TMessageFnProps) => {
+    const isBetweenWallets = sourceAccount.account_category === 'wallet' && targetAccount.account_category === 'wallet';
```

Comment:
could we rename it to isTransferBetweenWallets ?🤔

Reply:
Yup